packages/go/lab/loader.go

```diff
@@ -0,0 +1,110 @@
+package lab
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+
+	"github.com/specterops/bloodhound/dawgs/graph"
+)
+
+// GraphFixture is the JSON representation of the graph we are importing.
+type GraphFixture struct {
+	Nodes         []Node `json:"nodes"`
+	Relationships []Edge `json:"relationships"`
+}
+
+// Node is the JSON representation of a graph node.
+type Node struct {
+	// ID is the local identifier for the Node within the file.
+	// This ID is not preserved when imported into the database.
+	ID string `json:"id"`
+
+	// Labels are the node types. This is equivalent to what we call
+	// Kinds in BloodHound.
+	Labels []string `json:"labels"`
+
+	// Properties is the key:value map used for storing extra information
+	// on the Node. We currently do not validate that Nodes have an
+	// `object_id` property, but it is best practice to include one as
+	// `object_id` is the main identifier we use in BloodHound.
+	Properties map[string]any `json:"properties"`
+}
+
+// Edge is the JSON representation of a graph edge.
+type Edge struct {
+	// FromID is the local Node identifier for the start of an Edge.
+	FromID string `json:"fromId"`
+
+	// ToID is the local Node identifier for the end of an Edge.
+	ToID string `json:"toId"`
+
+	// Type is the 'label' we apply to the Edge. This is synonymous to
+	// the edge Kind in BloodHound.
+	Type string `json:"type"`
+
+	// Properties is the key:value map used for storing extra information
+	// on the Edge.
+	Properties map[string]any `json:"properties"`
+}
+
+// ParseGraphFixtureJsonFile takes in a fs.File interface and parses its contents into a
+// GraphFixture struct.
+func ParseGraphFixtureJsonFile(fh fs.File) (GraphFixture, error) {
```

Comment:
Is there a reason this isn't just part of the `LoadGraphFixtureFile` function and is exported right now? Seems like it would make sense to just fold it into the parent function until other usages are apparent

Reply:
I personally feel this is a premature abstraction, but feel free to override